cranelift/codegen/src/isa/pulley_shared/lower.isle

```diff
@@ -167,6 +167,20 @@
 (rule (lower (has_type $I64 (iadd a b)))
       (pulley_xadd64 a b))
 
+;;;; Rules for `isub` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(rule (lower (has_type $I8 (isub a b)))
+      (pulley_xsub32 a b))
+
+(rule (lower (has_type $I16 (isub a b)))
+      (pulley_xsub32 a b))
```

Comment:
Also double-check me on this, is this right?

Reply:
I don't think so, since (unlike addition) subtraction lets high bits (which should be invisible) move down to the low bits (and become visible), right?

Reply:
let's merge this and get further in spec tests, and eventually differential fuzzing, which will definitely catch any issues here